modules/nf-core/cobrameta/tests/main.nf.test

```diff
@@ -0,0 +1,123 @@
+                input[1] = Channel.of(
+                    [
+                        [ id:'test'],
+                        [
+                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/coverage/test1_contigs.coverage.stats.txt', checkIfExists: true)
+                        ]
+                    ]
+                )
+                input[2] = Channel.of(
+                    [
+                        [ id:'test'],
+                        [
+                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/tsv/test1_contigs_ids.tsv', checkIfExists: true)
+                        ]
+                    ]
+                )
+                input[3] = Channel.of(
+                    [
+                        [ id:'test'],
+                        [
+                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/bam/test1_contigs.sorted.bam', checkIfExists: true)
+                        ]
+                    ]
+                )
+                input[4] = 'metaspades'
+                input[5] = 27
+                input[6] = 127
+                """
+            }
+        }
+
+        then {
+            assertAll (
+                { assert process.success },
+                { assert snapshot(
+                    process.out.get(0).findAll { it != "log" }
+                    ).match()
+                },
+                { assert file(process.out.log.get(0).get(1)).exists() }
+            )
+        }
+    }
+
+    test("sarscov2 - bam - stub") {
+
+        options "-stub"
+
+        when {
+            process {
+                """
+                input[0] = GUNZIP.out.gunzip
+                input[1] = Channel.of(
+                    [
+                        [ id:'test'],
+                        [
+                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/coverage/test1_contigs.coverage.stats.txt', checkIfExists: true)
+                        ]
+                    ]
+                )
+                input[2] = Channel.of(
+                    [
+                        [ id:'test'],
+                        [
+                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/tsv/test1_contigs_ids.tsv', checkIfExists: true)
+                        ]
+                    ]
+                )
+                input[3] = Channel.of(
+                    [
+                        [ id:'test'],
+                        [
+                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/bam/test1_contigs.sorted.bam', checkIfExists: true)
+                        ]
+                    ]
+                )
+                input[4] = 'metaspades'
```

Comment:
I would denote the params in an extra section directly after when

```
params {
    assembler = 'metaspades'
    mink = 27
    maxk = 127
}
```

Then you can access those here like 

```suggestion
                input[4] = params.assembler
```
and so on.

Reply:
Ah and that also for the real test